File: src/test/java/com/github/qky666/selenidepom/test/java/mtp/pom/nopagefactorypom/ServicesShouldLoadRequiredErrorPage.java
package com.github.qky666.selenidepom.test.java.mtp.pom.nopagefactorypom;

import com.codeborne.selenide.SelenideElement;
import com.github.qky666.selenidepom.pom.Required;

import java.time.Duration;

import static com.codeborne.selenide.Condition.text;
import static com.codeborne.selenide.Selenide.$;

public class ServicesShouldLoadRequiredErrorPage extends MainFramePage {
    @Required
    public final SelenideElement principal() {return $("div.servicios-principal");}

    @Required
    public final SelenideElement title() {return $("h1.h2");}

    @Override
    public void customShouldLoadRequired(Duration timeout, String model, String lang) throws Throwable {
        super.customShouldLoadRequired(timeout, model, lang);
        title().shouldHave(text("Aseguramiento de la calidad con error"), timeout);
    }
}

File: src/test/kotlin/com/github/qky666/selenidepom/test/kotlin/mtp/pom/pages/services/ServicesShouldLoadRequiredErrorPage.kt
package com.github.qky666.selenidepom.test.kotlin.mtp.pom.pages.services

import com.codeborne.selenide.Condition.text
import com.codeborne.selenide.Selenide.element
import com.github.qky666.selenidepom.pom.Required
import com.github.qky666.selenidepom.test.kotlin.mtp.pom.common.MainFramePage
import java.time.Duration

class ServicesShouldLoadRequiredErrorPage : MainFramePage() {
    @Required val principal = element("div.servicios-principal")

    @Required val title = element("h1.h2")

    override fun customShouldLoadRequired(timeout: Duration, model: String, lang: String) {
        super.customShouldLoadRequired(timeout, model, lang)
        title.shouldHave(text("Text error"), timeout)
    }
}

val servicesShouldLoadRequiredErrorPage = ServicesShouldLoadRequiredErrorPage()
